src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -233,18 +242,21 @@ public ResponseEntity<byte[]> getDragItemFile(@PathVariable Long dragItemId, @Pa
     public ResponseEntity<byte[]> getFileUploadSubmission(@PathVariable Long exerciseId, @PathVariable Long submissionId, @PathVariable String filename,
             @RequestParam("access_token") String temporaryAccessToken) {
         log.debug("REST request to get file : {}", filename);
-        if (!validateTemporaryAccessToken(temporaryAccessToken, filename)) {
+
+        Claims requiredClaims = Jwts.claims();
+        requiredClaims.put(TokenProvider.EXERCISE_ID_KEY, exerciseId.intValue());
+        requiredClaims.put(TokenProvider.SUBMISSION_ID_KEY, submissionId.intValue());
+        requiredClaims.put(TokenProvider.FILENAME_KEY, TokenProvider.DOWNLOAD_FILE + filename);
```

Comment:
same below.
```suggestion
        var requiredClaims = Map.of(TokenProvider.EXERCISE_ID_KEY, exerciseId,
            TokenProvider.SUBMISSION_ID_KEY, submissionId,
            TokenProvider.FILENAME_KEY, TokenProvider.DOWNLOAD_FILE + filename
        );
```

Reply:
Changed requiredClaims to a Map object, also in the other occurences